### PR TITLE
Added cl_khr_fp16 extension support for test_vloadstore from basic

### PR DESCRIPTION
**Pull request by shajder**
According to issue description (basic section):

https://github.com/KhronosGroup/OpenCL-CTS/issues/142

### CODE REVIEW

File: test_conformance/basic/test_vloadstore.cpp [hunk 1]
```diff
@@ -360,53 +378,30 @@ int test_vload_local(cl_device_id device, cl_context context, cl_command_queue q
     cl_ulong localSize;
     int error = clGetDeviceInfo( device, CL_DEVICE_LOCAL_MEM_SIZE, sizeof( localSize ), &localSize, NULL );
     test_error( error, "Unable to get max size of local memory buffer" );
-    if( localSize > 10240 )
-        localSize = 10240;
-    if (localSize > 4096)
-        localSize -= 2048;
-    else
-        localSize /= 2;
+    localSize = std::min(localSize, (cl_ulong)2048);
```
Comment (EwanC) [inline test_conformance/basic/test_vloadstore.cpp:381]: Think this is a change to the max local memory tested, for [CL_DEVICE_LOCAL_MEM_SIZE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_LOCAL_MEM_SIZE) devices will probably report at least. Which would have been reduced to 8KB before, but now will be 2KB.

Were you trying to simplify the code with this chane, or was there a test performance reason behind it?

Same comment for the identical change in vstore.

Comment (shajder) [reply test_conformance/basic/test_vloadstore.cpp:381]: indeed I think I wanted to simplify those blurry conditions. Do you want me to restore original code ?

Comment (shajder) [reply test_conformance/basic/test_vloadstore.cpp]: @EwanC previous sizes reverted

**Review by EwanC — DISMISSED** [review]

**Review by svenvh — CHANGES_REQUESTED** [review]

File: test_conformance/basic/test_vloadstore.cpp [hunk 2]
```diff
@@ -145,28 +218,32 @@ int test_vload( cl_device_id device, cl_context context, cl_command_queue queue,
...
 
-
     // Create the reference results
-    memset( referenceBuffer, 0, numLoads * typeSize * vecSize * sizeof(cl_char));
+    referenceBuffer.assign(0, numLoads * typeSize * vecSize);
     for( i = 0; i < numLoads; i++ )
     {
-        memcpy( referenceBuffer + i * typeSize * vecSize, ( (char *)(void *)inBuffer ) + ( ( offsets[ i ] * vecSize ) + alignmentOffsets[ i ] ) * typeSize,
-               typeSize * vecSize );
+        memcpy(&referenceBuffer[i * typeSize * vecSize],
```
[the remark this answers is not in the document]
Comment (shajder) [reply test_conformance/basic/test_vloadstore.cpp:231]: Done, thanks

**Review by lakshmih — APPROVED** [review]

**Review by svenvh — DISMISSED** [review]

**Review by svenvh — APPROVED** [review]

**Comment by bcalidas** [comment]
Looks good.

**Pull request merged**